src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrInitializer.cs

```diff
@@ -64,6 +67,55 @@ public class VmrInitializer : VmrManagerBase, IVmrInitializer
             throw new EmptySyncException($"Repository {mapping.Name} already exists");
         }
 
+        var reposToUpdate = new Queue<(SourceMapping mapping, string? targetRevision, string? targetVersion)>();
```

Comment:
Do you plan to include also some tests? It'd be good to have the algorithm tested thoroughly.

Reply:
Yes, we're definitely adding tests.
I'm waiting on the library from Alex and also for the code here to settle first.
We already have tests for the core code that creates patches